full-service/src/db/account.rs

```diff
@@ -574,6 +580,35 @@ impl AccountModel for Account {
 
         Ok(highest_subaddress_index as u64 + 1)
     }
+
+    fn account_key(&self) -> Result<Option<AccountKey>, WalletDbError> {
+        if self.view_only {
+            return Ok(None);
+        }
+
+        let account_key: AccountKey = mc_util_serial::decode(&self.account_key)?;
+        return Ok(Some(account_key));
+    }
+
+    fn view_account_key(&self) -> Result<ViewAccountKey, WalletDbError> {
+        if self.view_only {
+            return Ok(mc_util_serial::decode(&self.account_key)?);
+        }
+
+        let account_key: AccountKey = mc_util_serial::decode(&self.account_key)?;
+        let view_account_key = ViewAccountKey::from(&account_key);
+        return Ok(view_account_key);
+    }
+
+    fn view_private_key(&self) -> Result<RistrettoPrivate, WalletDbError> {
+        if self.view_only {
+            let view_account_key: ViewAccountKey = mc_util_serial::decode(&self.account_key)?;
+            return Ok(*view_account_key.view_private_key());
+        }
+
+        let account_key: AccountKey = mc_util_serial::decode(&self.account_key)?;
+        return Ok(*account_key.view_private_key());
```

Comment:
Can this be simplified by calling `self.view_account_key()?.view_private_key()`?